### ISSUE TITLE
Roadmap to stable release

### ISSUE THREAD

**Issue by pradyunsg** [pradyunsg/furo#1]
This issue is meant to be a planning/dumping ground for tasks that have to be done prior to the first stable release of furo.

- [x] Write the documentation for this theme
  - [x] Finalize skeleton structure.
  - [x] Write the content!
- [x] Deploy the documentation w/ GitHub Pages
- [x] Stylize API documentation
- [x] Stylize captions
- [x] Stylize field lists (low priority)
- [x] Stylize the remaining permalinks
- [x] Switch to `em` for typography, instead of using `rem` everywhere
- [x] Tweak spacing for headings & paragraphs -- they might have a little too much margin?
- [x] Fix page's height on mobile, where 100%-100vh is not 0. -.-
- [x] Fix z-index levels for overlay (the toctree overlay needs to be above the sidebar and vice-versa).
- [x] Skeleton + styling for index pages (#50)
- [x] ~Add JS-based buttons to copy code blocks~
- [x] ~Create a good looking landing page for the documentation~

**Comment by pradyunsg** [comment]
https://github.com/tholman/github-corners is amazing and we should have that at least on the landing page for this theme's documentation (if not available on all pages).

**Comment by pradyunsg** [comment]
> I suspect you're already aware, but if not,

Even though I'm a collaborator on executablebooks, I didn't know about this. Thanks for the link! ^>^

**Comment by pradyunsg** [comment]
Actually, that doesn't *need* to be done here, or be critical path for the release. :)

**Comment by goodmami** [comment]
No, but it doesn’t work now because for some reason the `DOCUMENTATION_OPTIONS` object is not being set. I’m not at my computer now but I think it seemed like `base.html` was not being used? Although the “basic” theme also sets it so I’m not sure why it didn’t work. I can take a closer look tomorrow.

**Comment by pradyunsg** [comment]
@goodmami Yea, I was trying to reduce the number of assets that are loaded on each page. Turns out, a *lot* of stuff out there assumes how Sphinx works, and it's likely easier to just mimic it rather than trying to optimize things -- this was fixed in bea26d963c4c9fe1025bc734d04bc85cd52b1886.

**Comment by goodmami** [comment]
Oops, my fault. I was using the version from PyPI which doesn't have that commit. I withdraw my statements above about it not working.
I appreciate your goal to cut out the extra stuff. I hope the situation with Sphinx improves such that it can be realized.

**Comment by pradyunsg** [comment]
Okay, so there's lots of documentation to write now!

**Comment by bastimeyer** [comment]
@pradyunsg Just wanted to quickly ask if there's a chance for a beta20 release in the next couple of days. Is there anything that needs to be done first? The last release was 10 days ago and there have been some bugfixes since then which I'd like to have included in my docs builds. Thank you!

Btw, is there no release deploy/publish CI config?
- https://github.com/pradyunsg/furo/blob/b6c7c76bc1a5aa8f513eafbc5bd87b655a8e801f/.github/workflows/deploy.yml#L4
- https://github.com/pradyunsg/furo/blob/b6c7c76bc1a5aa8f513eafbc5bd87b655a8e801f/noxfile.py#L113

**Comment by pradyunsg** [comment]
Beta 20 has been uploaded.

@bastimeyer https://pradyunsg.me/furo/contributing/workflow/#release-process

**Comment by pradyunsg** [comment]
Just #50 is left. And, with that, I think I can move out of betas! :)

**Comment by pradyunsg** [comment]
I'll mark the releases I make starting next month as stable. :)

**Comment by jtpavlock** [comment]
Any chance of supporting Sphinx v4 with the next release?

**Comment by pradyunsg** [comment]
Yup, that's the rough plan. I just need to test it out, since I know that there have been some changes around the HTML output that themes have to work with.

https://github.com/pradyunsg/furo/discussions/148#discussioncomment-739767

**Comment by jtpavlock** [comment]
> Yup, that's the rough plan. I just need to test it out, since I know that there have been some changes around the HTML output that themes have to work with.
> 
> [#148 (comment)](https://github.com/pradyunsg/furo/discussions/148#discussioncomment-739767)

Missed that discussion, thanks!

**Comment by pradyunsg** [comment]
Okay, with Sphinx 4 settled down, we're basically there.

I have known for a while that I should write down a "policy document", describing the promises that this theme makes to the users; clarifying the versioning policy, change management policy and support policy. I've been procrastinating on that, but that should probably stay a blocker for "going stable".

**Comment by pradyunsg** [comment]
Annd... done!

https://pradyunsg.me/furo/stability/
https://pradyunsg.me/furo/changelog/#aspiring-avocado

:)
